src/librustc_mir/const_eval.rs

```diff
@@ -253,9 +250,6 @@ impl<'mir, 'tcx> interpret::Machine<'mir, 'tcx> for CompileTimeEvaluator {
                 ecx.goto_block(ret)?; // fully evaluated and done
                 return Ok(None);
             }
-            return Err(
```

Comment:
I had to remove this because the `is_const_fn` above is now overly restrictive (it will forbid calling `str::len` with the `const_str_len` feature gate but without the `const_slice_len` feature gate because that calls `[u8]::len` internally). We already check all the guarantees in the `const_qualif` pass, there's no need to check them again at evaluation time.